pyenttec/test.py

```diff
@@ -1,15 +1,48 @@
 """Tests for offline DMX port."""
-from nose.tools import assert_equal, assert_raises
-from . import DMXConnectionOffline, DMXAddressError
+from nose.tools import assert_equal, assert_raises, raises
+from . import DMXConnectionOffline, DMXAddressError, DMXOverflowError
+
+def get_test_port(univ_size=24):
+    return DMXConnectionOffline(univ_size=univ_size)
 
 def test_offline_port():
     univ_size = 24
-    port = DMXConnectionOffline(univ_size=univ_size)
+    port = get_test_port(univ_size)
 
     assert_equal(univ_size, len(port.dmx_frame))
     assert all(chan == 0 for chan in port.dmx_frame)
 
     port.set_channel(0, 1)
     assert_equal(1, port.dmx_frame[0])
 
-    assert_raises(DMXAddressError, port.set_channel, univ_size, 0)
\ No newline at end of file
+    assert_raises(DMXAddressError, port.set_channel, univ_size, 0)
+
+def test_port_item_accessors():
+    univ_size = 24
+    port = get_test_port(univ_size)
+    port[univ_size - 1] = 2
+    assert_equal(2, port.dmx_frame[univ_size - 1])
+    port[0] = 2
+    assert_equal(2, port.dmx_frame[0])
+
+@raises(DMXAddressError)
+def test_port_setitem_max_address():
```

Comment:
Also nicer to test more edge cases when probing for them.  Give this a try:

```python
def test_port_setitem_max_address():
    univ_size = 24
    port = get_test_port(univ_size)
    def assert_fail_set_index(i):
        with assert_raises(DMXAddressError):
            port[i] = 0
    assert_fail_set_index(-1)
    assert_fail_set_index(univ_size)
    assert_fail_set_index(univ_size + 1)
```

Reply:
I like the closure thing, but -1 is a valid index currently. I'm guessing we want to leave as is?

I can't actually see a reason for someone wanting to do that in the context of DMX... but I feel like it's pretty standard python functionality so it might just be less surprising to leave it as allowed.

I guess we could add a test to formalise it.

Reply:
Yep, sorry, this feedback was all pre-coffee this morning and of course -1 is entirely valid here.